pycytominer/cyto_utils/cells.py

```diff
@@ -613,7 +622,12 @@ def _compartment_df_generator(
             sql=f"select {cols} from {compartment} limit 1",
             con=self.conn,
         )
-        typeof_str = ", ".join([f"typeof({x})" for x in compartment_row1.columns])
+        all_columns = compartment_row1.columns
+        if features != "infer":
+            all_columns = [x for x in all_columns if x in features]
+            col = ", ".join(all_columns)
```

Comment:
I am not sure where this variable is used.

Reply:
Good catch! Should be clear now what's happening